sources/core/Stride.Core/Serialization/Binary/BinarySerializationWriter.cs

```diff
@@ -31,55 +32,109 @@ public override void Serialize(ref bool value)
             UnderlyingStream.WriteByte(value ? (byte)1 : (byte)0);
         }
 
-#pragma warning disable CS0618 // Type or member is obsolete
         /// <inheritdoc />
         public override unsafe void Serialize(ref float value)
         {
-            UnderlyingStream.Write(Unsafe.As<float, uint>(ref value));
+            unsafe
+            {
+                fixed (float* ptr = &value)
```

Comment:
You can use the same MemoryMarshal here as with the reads to avoid `fixed`.
Maybe it makes sense to create a helper generic method `Span<byte> MemoryMarshalAsBytes<T>(ref T value)` to save on space in all those methods.

Reply:
`Writer.Write` has support for float though, so you can just do 
```cs
public override unsafe void Serialize(ref float value)
{
    Writer.Write(value);
}
```

Reply:
cant we then just use a binarywriter

Reply:
Right, sorry, I haven't looked through your PR. I don't know the whole context but if it's not a breaking change yeah. Though it's likely that you'll have to adapt anything that depends on this, if you're up for the task go ahead :)

Reply:
Do you know why they had the native stream at all back then ? why didnt they use the binarywriter

Reply:
If you look at #1469 - I think the whole Native stream was to allow having a memory stream over unmanaged memory, which became a thing in .NET 5.
If we can fully replace this code with something that comes in the framework out of the box, let's do it.